src/avro2json.c

```diff
@@ -401,6 +417,15 @@ static int avro_value_to_json_t(const avro_value_t *value, json_t **json,
     const void *val;
     size_t size;
     CHECKED_EV(avro_value_get_fixed(value, &val, &size));
+
+    if (conf->ms_hadoop_logical_types && is_ms_hadoop_logical_type_guid(value, size)) {
+        char guid_val[37]; // Guid is formatted as 36 characters (32 nibbles plus 4 hyphens): xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx, and we need a null-terminator
+
+        sprintf(guid_val, GUID_FORMAT, GUID_ARG((char*)val));
```

Comment:
Why not `CHECKED_PRINTF`, like below?

Reply:
CHECKED_PRINTF calls fprintf, printing to a file descriptor

Reply:
Here I just build a string, sending it to the output on the next line (we produce JSON differently than CSV, using CHECKED_ALLOC)